src/apps/copo_dtol_submission/views.py

```diff
@@ -82,33 +82,50 @@ def get_samples_for_profile(request):
         direction = request.GET.get("order[0][dir]", "")
         search = request.GET.get("search", "")
         dir = 1
+
+        current_user = get_current_user()
+        profile = Profile().get_record(profile_id)
+        is_associated_project_type_checker = False
+
+        # Sometimes profile is None
+        if isinstance(profile, dict):
+            associated_profiles = profile.get("associated_type",[])
+            is_associated_project_type_checker = any(AssociatedProfileType.objects.filter(users=current_user, name=x.get("value","")) for x in associated_profiles)
+        
         if direction == "desc":
             dir = -1
+            
         samples = []
         if profile_id and profile_id != 'None':
-            profile_type = Profile().get_type(profile_id)
-            if profile_type:
-                type = ""
-                match profile_type:
-                    case "Aquatic Symbiosis Genomics (ASG)":
-                        type = "asg"
-                    case "European Reference Genome Atlas (ERGA)":
-                        type = "erga"
-                    case "Darwin Tree of Life (DTOL)":
-                        type = "dtol"
-                    case "Darwin Tree of Life Environmental Samples (DTOL_ENV)":
-                        type = "dtol_env"
-                if type:
-                    samples = Sample().get_dtol_from_profile_id(
+           profile_type = Profile().get_type(profile_id)
+            
+           if profile_type:
+              type = ""
+              
+              match profile_type:
+                  case "Aquatic Symbiosis Genomics (ASG)":
+                      type = "asg"
+                  case "European Reference Genome Atlas (ERGA)":
+                      type = "erga"
+                  case "Darwin Tree of Life (DTOL)":
+                      type = "dtol"
+                  case "Darwin Tree of Life Environmental Samples (DTOL_ENV)":
+                      type = "dtol_env"
+              if type:
+                 samples = Sample().get_dtol_from_profile_id(
                         profile_id, filter, draw, start, length, sort_by, dir, search, type)
-                # notify_frontend(msg="Creating Sample: " + "sprog", action="info",
-                #                     html_id="dtol_sample_info")
+                 # notify_frontend(msg="Creating Sample: " + "sprog", action="info",
+                 #                     html_id="dtol_sample_info")
+                 if filter == 'pending':
```

Comment:
Please also filter the samples for those sequencing centre user who is not from sequencing_centre with is_approval_required = True